src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionTelemetry.cs

```diff
@@ -60,38 +62,41 @@ protected override void OnEventCommand(EventCommandEventArgs command)
         [NonEvent]
         public long BeforeResolution(object hostNameOrAddress)
         {
-            // System.Diagnostics.Metrics part
-            NameResolutionMetrics.BeforeResolution(hostNameOrAddress, out string? host);
-
-            // System.Diagnostics.Tracing part
-            if (IsEnabled())
+            bool telemetryEnabled = IsEnabled();
+            if (telemetryEnabled)
             {
                 Interlocked.Increment(ref _lookupsRequested);
                 Interlocked.Increment(ref _currentLookups);
 
                 if (IsEnabled(EventLevel.Informational, EventKeywords.None))
                 {
-                    host ??= NameResolutionMetrics.GetHostnameFromStateObject(hostNameOrAddress);
+                    string host = GetHostnameFromStateObject(hostNameOrAddress);
 
                     ResolutionStart(host);
                 }
 
                 return Stopwatch.GetTimestamp();
             }
 
-            return 0;
+            return telemetryEnabled || NameResolutionMetrics.IsEnabled() ? Stopwatch.GetTimestamp() : 0;
         }
 
         [NonEvent]
-        public void AfterResolution(long? startingTimestamp, bool successful)
+        public void AfterResolution(object hostNameOrAddress, long? startingTimestamp, bool successful)
         {
             Debug.Assert(startingTimestamp.HasValue);
+            if (startingTimestamp == 0)
+            {
+                return;
+            }
 
-            if (startingTimestamp != 0)
+            TimeSpan duration = Stopwatch.GetElapsedTime(startingTimestamp.Value);
+
+            if (IsEnabled())
             {
                 Interlocked.Decrement(ref _currentLookups);
 
-                _lookupsDuration?.WriteMetric(Stopwatch.GetElapsedTime(startingTimestamp.Value).TotalMilliseconds);
+                _lookupsDuration?.WriteMetric(duration.TotalMilliseconds);
```

Comment:
This should be measured in seconds rather than milliseconds.

Reply:
That would be an unrelated breaking change which I want to avoid in this PR. If it's desirable, let's open an issue and discuss it there. Maybe there are other counters in System.Net.* where we want to switch to seconds.

Reply:
Ok, this is the existing event counter. Yes, it should stay as milliseconds.